From 8813e5eff0ab85e991d11f3e467a0350f2f0b5d5 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 14:38:28 -0400
Subject: [net] Add skb_get_hash_raw

Message-id: <1409668716-4814-17-git-send-email-sassmann@redhat.com>
Patchwork-id: 90645
O-Subject: [RHEL7.1 PATCH 16/24] net: Add skb_get_hash_raw
Bugzilla: 1107934
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

From: Tom Herbert <therbert@google.com>

Function to just return skb->rxhash without checking to see if it needs
to be recomputed.

Signed-off-by: Tom Herbert <therbert@google.com>
Acked-by: Eric Dumazet <edumazet@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 57bdf7f42be05640f8080b06844c94367ad1884b)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index d00a482..81af98c 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -774,6 +774,11 @@ static inline __u32 skb_get_rxhash(struct sk_buff *skb)
  return skb->rxhash;
 }
 
+static inline __u32 skb_get_hash_raw(const struct sk_buff *skb)
+{
+ return skb->rxhash;
+}
+
 static inline void skb_clear_hash(struct sk_buff *skb)
 {
  skb->rxhash = 0;
-- 
1.7.1